Module not found: Error: Can't resolve '@sentry/utils'

### Is there an existing issue for this?

- [X] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [X] I have reviewed the documentation https://docs.sentry.io/
- [X] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Self-hosted/on-premise

### Which SDK are you using?

@sentry/browser

### SDK Version

7.94.1

### Framework Version

React 18.2.0

### Link to Sentry event

_No response_

### SDK Setup

_No response_

### Steps to Reproduce

1. Use Yarn v4 with PnP
2. Build with @sentry/browser and @sentry/react

### Expected Result

No errors. v7.93.0 was fine.

### Actual Result

> Module not found: Error: Can't resolve '@sentry/utils' in '/yarn/global/cache/@sentry-internal-replay-canvas-npm-7.94.1-869cfbb282-10c0.zip/node_modules/@sentry-internal/replay-canvas/esm'

Can you elaborate on why yarn pnp is very broken? I'd bet there are more issues with node modules than with pnp if we want to dive into such discussion. Btw there are no node modules with pnp but zipped packages. Therefore, the build is more reproducible with pnp.

Such errors occur when packages require other packages but does not declare them as dependencies. That works with node modules since there's no isolation and everything is mixed together nbut not pnp. Basically that's what @jnoordsij said.

I managed to fetch full error now

```
ERROR in ../../../../../.yarn/berry/cache/@sentry-internal-replay-canvas-npm-7.94.1-869cfbb282-10c0.zip/node_modules/@sentry-internal/replay-canvas/esm/index.js 1:0-47
Module not found: Error: Can't resolve '@sentry/utils' in '/Users/user/.yarn/berry/cache/@sentry-internal-replay-canvas-npm-7.94.1-869cfbb282-10c0.zip/node_modules/@sentry-internal/replay-canvas/esm'
resolve '@sentry/utils' in '/Users/user/.yarn/berry/cache/@sentry-internal-replay-canvas-npm-7.94.1-869cfbb282-10c0.zip/node_modules/@sentry-internal/replay-canvas/esm'
  Parsed request is a module
  using description file: /Users/user/.yarn/berry/cache/@sentry-internal-replay-canvas-npm-7.94.1-869cfbb282-10c0.zip/node_modules/@sentry-internal/replay-canvas/package.json (relative path: ./esm)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      request is not managed by the pnpapi
        @sentry-internal/replay-canvas tried to access @sentry/utils, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
        Required package: @sentry/utils
        Required by: @sentry-internal/replay-canvas@npm:7.94.1 (via /Users/user/.yarn/berry/cache/@sentry-internal-replay-canvas-npm-7.94.1-869cfbb282-10c0.zip/node_modules/@sentry-internal/replay-canvas/esm/)
 @ ../../../../../.yarn/berry/cache/@sentry-browser-npm-7.94.1-85b21ca3c0-10c0.zip/node_modules/@sentry/browser/esm/index.js 14:0-62 14:0-62
 @ ./src/index.ts 2:0-49 11:27-41
```

--- a/packages/replay-canvas/package.json
+++ b/packages/replay-canvas/package.json
@@ -61,7 +61,8 @@
   "dependencies": {
     "@sentry/core": "7.94.1",
     "@sentry/replay": "7.94.1",
-    "@sentry/types": "7.94.1"
+    "@sentry/types": "7.94.1",
+    "@sentry/utils": "7.94.1"
   },
   "engines": {
     "node": ">=12"

